Documentation/design-docs/DotNetCore-SharedPackageStore.md

```diff
@@ -0,0 +1,216 @@
+# .NET Core - Shared Package Store
+
+## Introduction
+To enable sharing of assemblies among all machine-wide .NET Core applications, a centralized shared assembly store is needed. It enables applications to be trimmed of the shared assemblies and frees the apps from carrying them. This document will describe such an assembly store and its lookup that applications can take advantage of during development and their deployment. It also describes commands that will augment `dotnet` to compose shared package store entries and publishing mechanisms through which apps can filter assemblies to reduce disk usage.
+
+### Packages Store
+
+The package store can be either a global system-wide folder or a user local folder:
+
++ **Global**:
+    - The `dotnet` root location -- on Windows, the folder is located in `C:\Program Files (x86)\`. See layout below.
+    - In the `packages/install` folder we expect packages to be installed ONLY through platform installers like MSI, pkg, deb, apt-get etc.
+    - The package layout composed with the `dotnet cache` command (details follow) are expected to be unzipped directly into the `packages` folder. Note, that this unzip step is a manual action.
+
+```
+    - dotnet.exe
+    - shared
+        - netcoreapp2.0
+            + 2.0.0-preview2-00001
+    - packages
+        - install
+            + refs
+            + netcoreapp2.0
+            + netcoreapp2.1
+        + refs
+        + netcoreapp2.0
+        + netcoreapp2.1  
+```
+
+The layout within `netcoreapp*` folders is a NuGet cache layout.
+
++ **User**: The user local location can be used to override the global location when probing assemblies and uses a very similar layout, except that the `packages/install` folder is NOT supported in the user profile. It is only supported in the global location.
+
+```
+    - %USERPROFILE% or $HOME/
+        - .dotnet
+          - {arch=x86|x64}
+            - packages 
+                = do =
+```
+
+**OPEN**: Should we have a `dotnet` relative location as well, if operating out of a zip installation of .NET Core binaries?
+
+### Composing a runtime (non-ref) package store
+
+To compose the layout of the shared package store, we will use a dotnet command called `dotnet cache`. We expect the *hosting providers* (ex: Antares) to use the command to prime their machines and framework authors who want to provide *pre-optimized package archives* create the compressed archive layouts.
+
+The layout is composed from a list of package names and versions specified as JSON: 
+
+**Roslyn Example**
+```xml
+ <ItemGroup>
+    <PackageReference Include="Microsoft.CodeAnalysis" Version="1.3.2" />
+    <PackageReference Include="Microsoft.CodeAnalysis.VisualBasic" Version="1.3.2" />
+    <PackageReference Include=""Microsoft.CodeAnalysis.VisualBasic.Features" Version="1.3.2" />
+  </ItemGroup>
+```
+
```

Comment:
Is this really the only xml needed or does it have to be an MSBuild file? If so, does it need to reference a specific SDK?